Druid-Tests/DRDispatchTableBuilderTest.class.st

```diff
@@ -23,16 +23,19 @@ DRDispatchTableBuilderTest >> setUp [
 
 	super setUp.
 	interpreter :=	DRInterpreterToCompiler fromInterpreterClass: DRBasicCogInterpreterArithmeticPrimitives.
-	dispatchTableBuilder := DRCogitDispatchTableBuilder primitiveTableMethodFrom: self compilationUnitForTesting
+	dispatchTableBuilder := DRCogitDispatchTableBuilder fromCompilationUnit: self compilationUnitForTesting
 ]
 
 { #category : #tests }
 DRDispatchTableBuilderTest >> testCollectAllMethodsOn [
 
 	| pcgMethodNodes |
-	self assert: dispatchTableBuilder initPrimitiveTable isEmpty.
+	
+	self assertEmpty: dispatchTableBuilder initPrimitiveTable.
+	self assertEmpty: dispatchTableBuilder initBytecodeTable.
+	
 	pcgMethodNodes := dispatchTableBuilder collectAllMethodsOn: self jitCompilerClassForTest class.
 	self
 		assertCollection: (pcgMethodNodes collect: #selector)
-		hasSameElements: #(primitiveTableArray initializePrimitiveTable)
+		hasSameElements: #(primitiveTableArray initializePrimitiveTable initializeBytecodeTableForSistaV1)
```

Comment:
Hmm, making a step backwards, this looks a bit entangled, we should discuss the general architecture.